From 930593c3bdabb70c7af0bd44586db17288fa6629 Mon Sep 17 00:00:00 2001
From: b07211 <b07211@freescale.com>
Date: Wed, 15 Sep 2010 17:39:58 -0400
Subject: [PATCH 076/635] ENGR00126251 Camera/Camcorder:switch Camcorder to
 camera cause black screen.

Add a init the variable video_frames in v4l init
Signed-off-by: b07211 <b07211@freescale.com>
---
 liboverlay/overlay.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/liboverlay/overlay.cpp b/liboverlay/overlay.cpp
index 0aff876..fbce82a 100755
--- a/liboverlay/overlay.cpp
+++ b/liboverlay/overlay.cpp
@@ -765,7 +765,7 @@ static int overlay_init_v4l(struct overlay_control_context_t *dev)
     	    OVERLAY_LOG_ERR("Error!VIDIOC_CROPCAP getting failed for dev %s",V4L_DEV_NAME);
             return -1;
         }
-
+        dev->video_frames = 0;
 
         struct v4l2_crop crop;
         /* set the image rectangle of the display by 
@@ -1009,7 +1009,13 @@ static void overlay_destroyOverlay(struct overlay_control_device_t *dev,
     }
     
     if(instance < MAX_OVERLAY_INSTANCES) {
-        OVERLAY_LOG_INFO("Destory the overlay instance id %d",instance);
+        OVERLAY_LOG_INFO("****Destory the overlay instance id %d",instance);
+        //Flush the buffer in queue
+        overlay_data_shared_t *dataShared = ctx->overlay_intances[instance]->mDataShared;
+        if((dataShared != NULL)&&(dataShared->queued_count > 0)) {
+            OVERLAY_LOG_ERR("Error!Still %d buffer in queue",dataShared->queued_count);
+        }
+
         ctx->overlay_number--;
         ctx->overlay_instance_valid[instance] = 0;
         ctx->overlay_intances[instance] = NULL;
-- 
1.8.0

